### PR TITLE
detect-filemd5: convert unittests to FAIL/PASS APIs

### PR DESCRIPTION
**Pull request by AkakiAlice**
Task: #4035

Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/4035

Previous PR: https://github.com/OISF/suricata/pull/8050

Describe changes:
- Fix code style error.
- Fix wrong use of macro.
- Fix compilation errors.

### CODE REVIEW

**Comment by AkakiAlice** [comment]
Hello I'm trying to understand why test "builds / Fedora 36 (debug, clang, asan, wshadow, rust-strict)" failed, I read the error message but I couldn't understand. I also ran the test that I converted and I ran suricata-verify but I'm using Ubuntu 22.04 so I couldn't reproduce the error. If anyone could help me i would be grateful

**Comment by inashivb** [comment]
> Hello everyone! I'm trying to understand why test "builds / Fedora 36 (debug, clang, asan, wshadow, rust-strict)" failed, I read the error message but I couldn't understand. I also ran the test that I converted and it worked, I ran suricata-verify but I'm using Ubuntu 22.04 so I couldn't reproduce the error. If anyone could help me i would be grateful

Hello! :)
That failure seemed unrelated to your changes so I ran that job again and it seems to have passed so you are good.

File: src/detect-filemd5.c [hunk 1]
```diff
@@ -78,57 +78,38 @@ static int DetectFileMd5Setup (DetectEngineCtx *de_ctx, Signature *s, const char
...
-        if (ptr == NULL)
-            return 0;
-        else
-            return 1;
-    }
-    return 0;
+    FAIL_IF_NOT(ReadHashString(md5, string, "file", 88, 32) == 1);
+    void *ptr = ROHashLookup(hash, &md5, (uint16_t)sizeof(md5));
+    FAIL_IF_NULL(ptr);
+    PASS;
```
Comment (jufajardini) [inline src/detect-filemd5.c:84]: I don't think that this one needs conversion, as it's a utility function for the unittest itself.

(Thinking about it, I'd say that a good way to ponder about if a function should be converted or not would be to check what the function that registers the unittests has in it...)

**Review by jufajardini — CHANGES_REQUESTED** [review]
Unit test conversion looks good to me, but I don't think we should convert `MD5MatchLookupString()`

I'd say that if you revert those changes, this contribution would be good to go :)

**Pull request closed without merging**